client/src/App.js

```diff
@@ -303,8 +331,50 @@ export default class NormalDisplay extends React.Component {
                 </pre>
     }
 
+    async checkJournalReady(journalUrl) {
```

Comment:
I would combine these two api calls, `getJournalUrl` and `checkJournalReady` and just have `checkJournalReady`. The backend can take care of getting the journal URL (it actually already has it through calling register) and then use that to check if the journal is ready, without having to store the journalUrl in the frontend

Reply:
thanks! Excuse me, should have cleaned it up.